MovieGPT-api/src/main/java/lt/techin/group/project/security/AuthController.java

```diff
@@ -2,59 +2,33 @@
-import lt.techin.group.project.rest.dto.UserDto;
+import lt.techin.group.project.service.UserService;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.security.authentication.AuthenticationManager;
-import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
-import org.springframework.security.core.Authentication;
-import org.springframework.security.core.context.SecurityContextHolder;
-import org.springframework.security.core.userdetails.UserDetails;
-import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.Set;
+import java.util.Map;
 
 @AllArgsConstructor
 @RestController
 @RequestMapping("/auth")
 public class AuthController {
 
-    private AuthenticationManager authenticationManager;
-    private JwtService jwtService;
-    private UserRepository userRepository;
-    private PasswordEncoder passwordEncoder;
+    private UserService userService;
 
 
     @PostMapping("/login")
-    public String login(@RequestBody @Valid UserLoginRequest userLoginRequest) {
-        Authentication authentication = authenticationManager.authenticate(
-                new UsernamePasswordAuthenticationToken(userLoginRequest.getUsername(), userLoginRequest.getPassword()));
-        SecurityContextHolder.getContext().setAuthentication(authentication);
-
-        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
-        UserDto userDto = new UserDto(userRepository.findByUsername(userDetails.getUsername()));
-        return jwtService.generateToken(userDto);
+    public ResponseEntity<Map<String, String>> login(@RequestBody @Valid UserLoginRequest userLoginRequest) {
+        return ResponseEntity.status(HttpStatus.OK).body(userService.login(userLoginRequest));
     }
-    @PostMapping("/signup")
-    public ResponseEntity<UserDto> signup(@RequestBody @Valid UserSignupRequest userSignupRequest) {
-
-        if (userRepository.findByUsername(userSignupRequest.getUsername()) != null) {
-            return ResponseEntity.badRequest().body(new UserDto(userRepository.findByUsername(userSignupRequest.getUsername())));
-        }
-        User newUser = new User();
-        newUser.setUsername(userSignupRequest.getUsername());
-        newUser.setPassword(passwordEncoder.encode(userSignupRequest.getPassword()));
-        newUser.setEmail(userSignupRequest.getEmail());
-        newUser.setRoles(Set.of(Roles.USER));
-        userRepository.save(newUser);
-
-        return ResponseEntity.ok(new UserDto(newUser));
 
+    @PostMapping("/signup")
+    public ResponseEntity<Map<String, String>> signup(@RequestBody @Valid UserSignupRequest userSignupRequest) throws UsernameAlreadyExistsException, EmailAlreadyExistsException {
+        return ResponseEntity.status(HttpStatus.CREATED).body(userService.register(userSignupRequest));
```

Comment:
For signup to also login the user in the front, can you make it so sigunp still returns token somewhere?

Reply:
I think better in front after sign up in same function, call login endpoint and this way get token. 
Because beside giving just token, we need approve user and pass it through various security points.
I can make so after signup it calls login method, but i don't this is good practice to do it in back end.